src/lib.rs

```diff
@@ -91,25 +99,77 @@ fn get_output_mode(args: &Args) -> OutputMode {
     }
 }
 
+struct MaybeInitializedCaptureIndex(AtomicU32);
+
+impl MaybeInitializedCaptureIndex {
+    fn mark_failed(&self) {
+        self.0.store(u32::MAX - 1, Ordering::Relaxed);
```

Comment:
What I did (I guess in order to avoid having multiple atomics going on at which point the whole `Ordering` thing gets harder to reason about) was just have a single `AtomicUsize` and encode the "uninitialized" and "invalid capture name" states into its value